src/core/stdlib.cpp

```diff
@@ -0,0 +1,126 @@
+            std::regex pattern_number (R"(\s*\d+\.?\d*)");
+            std::regex pattern_hex (R"(\s*0[xX][\dA-Fa-f]+\.?[\dA-Fa-f]*)");
+            std::regex pattern_exp (R"(\s*\d+\.?\d*[eE]-?\d+)");
+
+            if(std::regex_match(number.value, pattern_number) || std::regex_match(number.value, pattern_hex) || std::regex_match(number.value, pattern_exp)){
+                return std::stod(number.value);
+            }else{
+                return Nil();
+            }
+        },
+        [](const String& number, Number base) -> Value {
+            //match again with pattern, but this time with 1 .
+            if(base < 2 || base > 36){
+                throw std::runtime_error("base is to high. base must be >= 2 and <= 36");
+            }else{
+                std::regex pattern_number (R"(\s*\d+\.\d*)");
+                std::regex pattern_hex (R"(\s*0[xX][\dA-Za-z]+\.[\dA-Za-z]*)");
+                std::regex pattern_exp (R"(\s*\d+\.\d*[eE]-?\d+)");
+                //parse number to double
+                if(std::regex_match(number.value, pattern_number) || std::regex_match(number.value, pattern_hex)){
+                    auto parts = split_string(number.value, '.');
+                    int precomma = std::stoi(parts.first, nullptr, base.value);
+                    int postcomma = std::stoi(parts.second, nullptr, base.value);
+                    return precomma + postcomma * std::pow(base.value, parts.second.size());
+                }else if (std::regex_match(number.value, pattern_exp)) {
+                    auto number_exp = split_string(number.value, 'e');
+                    int exp = std::stoi(number_exp.second);
+                    auto parts = split_string(number_exp.first, '.');
+                    int precomma = std::stoi(parts.first, nullptr, base.value);
+                    int postcomma = std::stoi(parts.second, nullptr, base.value);
+                    double number_res = precomma + precomma + postcomma * std::pow(base.value, parts.second.size());
+                    return number_res * std::pow(base.value, exp);
+                }else{
+                    return Nil();
+                }
+            }
+        },
+        [](Number number, Nil /*unused*/) -> Value {
+            return number;
+        },
+        [](auto  /*a*/, auto  /*b*/) -> Value { return Nil();}
+    }, number.raw(), base.raw());
+}
+
+auto type(const CallContext& ctx) -> Value {
+    auto v = ctx.arguments().get(0);
+
+    //TODO: Change return to static variable of Struct of type
+    return std::visit(overloaded{
+        [](Bool /*b*/) -> Value { return "boolean"; },
+        [](Number  /*n*/) -> Value { return "number"; },
+        [](const String&  /*s*/) -> Value { return "string"; },
+        [](const Table&  /*t*/) -> Value { return "table"; },
+        [](const Function&  /*f*/) -> Value { return "function"; },
+        [](Nil  /*nil*/) -> Value { return "nil"; }
+        //TODO: add type for metatables
+    }, v.raw());
```

Comment:
```suggestion
    return std::visit(
        [](auto value) -> Value { return std::string(value.TYPE); },
        v.raw());
```

The `TYPE` static member variable is available on all `Value` types so this should work.